### PR TITLE
Add createTimestamp to REST service

### LINKED ISSUE #14009: CreatedTimestamp on REST import not used
### Description

The CreatedTimestamp (if set on the user rest creation call) is always discarded in favor of `Time.currentTimeMillis()`. This shouldn't be the case for scenarios when someone is REST importing from a legacy system.

### Discussion

_No response_

### Motivation

Handle graceful import of users from legacy systems.

### Details

From `MapUserProvider`.

image

### PR DESCRIPTION
**Pull request by heddn**
Fixes #14009.

Untested as I don't really have a good way to compile and test locally, but from looking at the code changes here, this _should_ fix the issue.

### CODE REVIEW

File: services/src/main/java/org/keycloak/services/resources/admin/UserResource.java [hunk 1]
```diff
@@ -268,6 +268,7 @@ public static void updateUserFromRep(UserProfile profile, UserModel user, UserRe
 
         if (rep.isEnabled() != null) user.setEnabled(rep.isEnabled());
         if (rep.isEmailVerified() != null) user.setEmailVerified(rep.isEmailVerified());
+        if (rep.getCreatedTimestamp() != null) user.setCreatedTimestamp(rep.getCreatedTimestamp());
```
Comment (pedroigor) [inline services/src/main/java/org/keycloak/services/resources/admin/UserResource.java:271]: The problem here is that the `updateUserFromRep` is called also during updates. We don't want to set this attribute but only during creation (or importing).

**Review by pedroigor — CHANGES_REQUESTED** [review]
For testing, you can look at [this test class](https://github.com/keycloak/keycloak/blob/main/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/admin/UserTest.java). There are similar tests there.

File: services/src/main/java/org/keycloak/services/resources/admin/UserResource.java (hunk 1 above)
Comment (heddn) [reply services/src/main/java/org/keycloak/services/resources/admin/UserResource.java:271]: I don't see why we shouldn't let a REST api add a value to this. We are checking for non null.

Comment (pedroigor) [reply services/src/main/java/org/keycloak/services/resources/admin/UserResource.java:271]: It surely does. But during creation not during updates. Making sure this attribute is only set during creation solves your problem, right?

Comment (heddn) [reply services/src/main/java/org/keycloak/services/resources/admin/UserResource.java:271]: I guess. But I'm also not sure why a later REST call in an update to (fix?) the create timestamp should be dis-allowed.

Comment (heddn) [reply services/src/main/java/org/keycloak/services/resources/admin/UserResource.java:271]: 🤔 how do I go about discerning if this is a create vs an update?

Comment (heddn) [reply services/src/main/java/org/keycloak/services/resources/admin/UserResource.java]: I did add a change to only do on creations and added a test case.

**Review by pedroigor — APPROVED** [review]
@heddn Thanks. 

@mposolda Does it work for you?

**Review by mposolda — APPROVED** [review]

**Pull request merged**